cmake/OpenCVDetectCUDA.cmake

```diff
@@ -414,8 +414,25 @@ if(CUDA_FOUND)
     endforeach()
   endmacro()
 
+  macro(ocv_check_windows_crt_linkage)
+    # The new MSVC runtime abstraction is only useable if CUDA is a first class language
+    if(WIN32 AND POLICY CMP0091)
+      cmake_policy(GET CMP0091 MSVC_RUNTIME_SET_BY_ABSTRACTION)
+      if(MSVC_RUNTIME_SET_BY_ABSTRACTION STREQUAL "NEW")
+        if(BUILD_SHARED_LIBS)
+          set(CMAKE_CXX_FLAGS_RELEASE ${CMAKE_CXX_FLAGS_RELEASE} " /MD")
+          set(CMAKE_CXX_FLAGS_DEBUG ${CMAKE_CXX_FLAGS_DEBUG} " /MDd")
+        else()
+          set(CMAKE_CXX_FLAGS_RELEASE ${CMAKE_CXX_FLAGS_RELEASE} " /MT")
```

Comment:
Sometimes I need to compile OpenCV as static library but still needs it to be compiled with dynamic CRT.
I was using `BUILD_WITH_STATIC_CRT`.  So it is perhaps better to use this flag than `BUILD_SHARED_LIBS`?

So I am not sure how the NEW logic is supposed to handle the 4 cases on Windows
- `BUILD_SHARED_LIBS=OFF` `BUILD_WITH_STATIC_CRT=OFF`
- `BUILD_SHARED_LIBS=OFF` `BUILD_WITH_STATIC_CRT=ON`
- `BUILD_SHARED_LIBS=ON` `BUILD_WITH_STATIC_CRT=OFF`
- `BUILD_SHARED_LIBS=ON` `BUILD_WITH_STATIC_CRT=ON`

Reply:
As you mention I missed that the existing logic 
```
if(NOT BUILD_SHARED_LIBS AND BUILD_WITH_STATIC_CRT)
  dynamic CRT
else
  static CRT
```
from https://github.com/opencv/opencv/blob/588ddf1b181aa7243144b27d65fc7690fb89e344/cmake/OpenCVCRTLinkage.cmake#L36

allows 
> BUILD_SHARED_LIBS=OFF BUILD_WITH_STATIC_CRT=OFF

and doesn't respect

> BUILD_SHARED_LIBS=ON BUILD_WITH_STATIC_CRT=ON

To be consistent shouldn't the new logic be the same i.e.

```
 if(MSVC_RUNTIME_SET_BY_ABSTRACTION STREQUAL "NEW")
    if(NOT BUILD_SHARED_LIBS AND BUILD_WITH_STATIC_CRT)
      set(CMAKE_CXX_FLAGS_RELEASE ${CMAKE_CXX_FLAGS_RELEASE} " /MT")
      set(CMAKE_CXX_FLAGS_DEBUG ${CMAKE_CXX_FLAGS_DEBUG} " /MTd")
    else()
      set(CMAKE_CXX_FLAGS_RELEASE ${CMAKE_CXX_FLAGS_RELEASE} " /MD")
      set(CMAKE_CXX_FLAGS_DEBUG ${CMAKE_CXX_FLAGS_DEBUG} " /MDd")
```